modules/gdscript/gdscript_utility_functions.cpp

```diff
@@ -122,21 +122,26 @@ struct GDScriptUtilityFunctionsDefinitions {
 			} break;
 			case 1: {
 				VALIDATE_ARG_NUM(0);
-				int count = *p_args[0];
-				Array arr;
+				TypedArray<int64_t> arr;
+				int64_t count = *p_args[0];
 				if (count <= 0) {
 					*r_ret = arr;
 					return;
 				}
-				Error err = arr.resize(count);
+				if (count > (int64_t)std::numeric_limits<int>::max()) {
+					r_error.error = Callable::CallError::CALL_ERROR_INVALID_METHOD;
+					*r_ret = RTR("The given range has too many elements!");
+					return;
+				}
+				Error err = arr.resize((int)count);
 				if (err != OK) {
 					r_error.error = Callable::CallError::CALL_ERROR_INVALID_METHOD;
-					*r_ret = Variant();
+					*r_ret = RTR("Failed to resize the array!");
```

Comment:
Does this mean that `range()` can return a String? That seems odd...

Reply:
AFAIU the call will fail and this value won't be used as a return value of the `range`, it will be used just for the error. I'm not super familiar with this myself (hence requested **vnen**'s review) but erroring like this is already made in case of a step being zero:
https://github.com/godotengine/godot/blob/604493eb6ee3def65aca089c1d4bc71ea03ca19a/modules/gdscript/gdscript_utility_functions.cpp#L175-L179

In v4.0.beta17.official [c40020513]:
![Godot_v4 0-beta17_win64_0hrfbeYZ2L](https://user-images.githubusercontent.com/9283098/216679776-03334476-825c-4a8f-b621-e61530817469.png)
After unpausing the execution still only `hello` is in the output.

The returned string is being interpreted like that specifically in here:
https://github.com/godotengine/godot/blob/18e827ee46b00f6909c8c3b56bcd07d332b89cdc/modules/gdscript/gdscript_vm.cpp#L2147-L2159

I agree that's kinda strange how it's made. But it is what it is. So it seemed strange to me that only one specific case (step=0) was erroring like that. If it's possible to do so, why not specify the error in each case? Hence I've changed it like that.